client/src/main/scala/org/http4s/client/middleware/Logger.scala

```diff
@@ -0,0 +1,78 @@
+package org.http4s
+package client
+package middleware
+
+import cats.effect._
+import fs2._
+import org.http4s.util.CaseInsensitiveString
+import org.log4s.{Logger ⇒ SLogger}
+
+/**
+  * Simple Middleware for Logging All Requests and Responses
+  */
+object Logger {
+  def apply[F[_]: Effect](
+      logHeaders: Boolean,
+      logBody: Boolean,
+      redactHeadersWhen: CaseInsensitiveString => Boolean = Headers.SensitiveHeaders.contains
+  )(client: Client[F]): Client[F] =
+    Client.fromHttpService(
```

Comment:
I'm not sure whether this is the best approach here. It seems to work ok for low-volume clients but the to/from HttpService round-trip definitely adds some overhead. I'm also not sure how to implement this on `master`, which has deprecated `HttpService[F]`. I'd love suggestions on how to remove the `HttpService[F]` interface from these client classes and just deal with `Client[F]`.

Reply:
It seems highly likely that a client will be based on either `Http[Stream[F, ?], F]` or `Http[Resource[F, ?], F]` in the future.  And then the logging middleware can be built on `Http[G, F]` and be unified for servers and clients.